files-csharp/FastestWayToReadATextFileInCsharp/FastestWayToReadATextFileInCsharp/WaysToReadATextFileInCsharp.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Text;
+
+namespace FastestWayToReadATextFileInCsharp
+{
+    public class WaysToReadATextFileInCsharp
+    {
+        private readonly string _sampleFilePath;
+
+        public WaysToReadATextFileInCsharp(string sampleFilePath)
+        {
+            _sampleFilePath = sampleFilePath;
+        }
+
+        public string UseFileReadAllLines()
+        {
+            var stringArray = File.ReadAllLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseFileReadAllText() => File.ReadAllText(_sampleFilePath);
+        
+        public string UseFileReadLines()
+        {
+            var stringArray = File.ReadLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseStreamReaderReadLine()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+            var stringBuilder = new StringBuilder();
+            string fileLine;
+
+            while ((fileLine = streamReader.ReadLine()) != null)
+            {
+                stringBuilder.AppendLine(fileLine);
+            }
+
+            return stringBuilder.ToString().TrimEnd();
+        }
+
+        public string UseStreamReaderReadToEnd()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+
+            return streamReader.ReadToEnd();
+        }
+
+        public string UseStreamReaderReadBlock()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+            var buffer = new char[1024];
```

Comment:
@cjaynjoku For the span suggestion, can you please keep this with the array, and add one more method with the Span, this can affect performance. 
Add also these buffer size detail in the article and justify why we chose 4096 as @jshergal  suggested.